### ISSUE TITLE
Feature Request: Add "Never Overdue" to Product Details in Edit Product.

### ISSUE THREAD

**Issue by happyshow** [grocy/grocy#2262]
I know about the option to add -1 to Default due days in a product's details but it hasn't been working. They only way I can change a product to Never Overdue is to open it in Inventory, increase the count, check the box Never Overdue, save. Then open the product in Inventory again and reduce the number back to the actual number, check the box Never Overdue and save. Only after that, do certain products show as Never for Next due date in the Stock Overview.

A check box for Never Overdue in the product details would be great.

Product type needed for is my collection of alcohol (ie Whiskey, Burbon, Rum, etc)

**Comment by berrnd** [comment]
> add -1 to Default due days in a product's details but it hasn't been working

When that's set like that, "never overdue" (which is an alias for the date `2999-12-31`) is prefilled on [/purchase](https://demo.grocy.info/purchase) (or you can of course also set it manually there). What exactly is not working where? Please provide a reproducible example on a [persistent demo instance](https://www.reddit.com/r/grocy/comments/jwnk9b).

&nbsp;
> A check box for Never Overdue in the product details would be great.

I don't really get what "product details" refers to, but you can edit any stock entry on [/stockentries](https://demo.grocy.info/stockentries) and there the corresponding edit dialog of course also has the "Never overdue" shorthand checkbox.

&nbsp;
Maybe also have a look at the available [input shorthands for date fields](https://github.com/grocy/grocy#input-shorthands-for-date-fields) - the "never overdue"-date can also be set on any date input field by simply entering `x`.

&nbsp;
In conclusion I don't see that it's not possible to set "never overdue" anywhere. "Never overdue" is also always only an alias (for the aformentioned date `2999-12-31` for any date input and `-1` for the product option "Default due days"). So the checkbox just sets the corresponding value, there is no more magic behind.

The better place for general usage questions is the [r/grocy subreddit](https://www.reddit.com/r/grocy) as [mentioned in README](https://github.com/grocy/grocy#questions--help--bug-reports--feature-requests) and on the request template.

**Comment by happyshow** [comment]
Product details in inside of "Edit Product"

**Comment by berrnd** [comment]
As mentioned before, adding such a checkbox there (which will not be done) would technically not do anything else as setting that value to `-1`. As also just mentioned before, that checkbox on any other place just sets the corresponding "alias value", nothing more.

So when setting _"Default due day"_ to `-1` doesn't prefills "never overdue" on [/purchase](https://demo.grocy.info/purchase), then please simply provide a reproducible example how to get that. I just tested it and use it personally daily, it works.

I can imagine that exactly that is a problem when running v3.3.2 with PHP 8.1 or PHP 8.2 (which is unsupported, [README ref](https://github.com/grocy/grocy/tree/v3.3.2#how-to-install), and produces many errors related to number/string comparison stuff). But you haven't mentioned any little about anything, so guessing is the only option here.

**Comment by happyshow** [comment]
I see the checkbox in /stockentries . Thank you. Is there anywhere in the GUI to access /stockentries? I am new to Grocy. Sorry for the dumb questions.

**Comment by berrnd** [comment]
> Is there anywhere in the GUI to access /stockentries?

Of course, no single page is hidden / not linked anywhere - that button on [/stockoverview](https://demo.grocy.info/stockoverview):

![grafik](https://github.com/grocy/grocy/assets/2161815/6c2b695d-ba15-4646-9cf0-c1ad99dc4c19)

That context menu entry of each and every row on [/stockoverview](https://demo.grocy.info/stockoverview):

![grafik](https://github.com/grocy/grocy/assets/2161815/89f7166c-1e62-44f2-9200-ff91d5c90b94)

That button on the product card:

![grafik](https://github.com/grocy/grocy/assets/2161815/42c4f3a6-2d76-4f90-83cf-8d101151237e)

&nbsp;
> I am new to Grocy. Sorry for the dumb questions.

This here is the tracker for reproducible Bugs and Feature Requests, not the forum for any little usage question. Please note what's [mentioned in README](https://github.com/grocy/grocy#questions--help--bug-reports--feature-requests) and don't report any little thing you can't explain at a first sight as a Bug or Feature Requests - it's most likely not any of that and that's naturally the case when doing the first steps with a (kind of complex) software never used before.

**Comment by happyshow** [comment]
Thank you very much for your quick support. Forgive me for being naive as I'm learning Grocy.
